lib/socket.c

```diff
@@ -606,10 +605,18 @@ smb2_connect_async(struct smb2_context *smb2, const char *server,
                         return -1;
                 }
                 *str = 0;
+                port = str + 1;
+        }
+
+        port = strchr(port, ':');
+        if (port != NULL) {
+                *port++ = 0;
+        } else {
+                port = "445";
```

Comment:
Why string?

Reply:
A string and passing it to getaddrinfo() allows using service names and not just numbers.
I.e. /etc/services